src/modules/storage/LedgerStorage.ts

```diff
@@ -1682,6 +1747,117 @@ export class LedgerStorage extends Storages {
+                            proposalData.vote_start_height.toString(),
+                            proposalData.vote_end_height.toString(),
+                            proposalData.doc_hash.toBinary(Endian.Little),
+                            proposalData.fund_amount.toString(),
+                            proposalData.proposal_fee.toString(),
+                            proposalData.vote_fee.toString(),
+                            proposalData.tx_hash_proposal_fee.toBinary(Endian.Little),
+                            proposalData.proposer_address.toString(),
+                            proposalData.proposal_fee_address.toString(),
+                            ProposalStatus.ONGOING,
+                            DataCollectionStatus.PENDING,
+                        ]
+                    )
+                    .then(() => {
+                        resolve();
+                    })
+                    .catch((err) => {
+                        reject(err);
+                    });
+            });
+        }
+
+        function putPayload(storage: LedgerStorage, block_height: Height, tx_hash: Hash, tx: Transaction) {
+            return new Promise<void>(async (resolve, reject) => {
+                try {
+                    let buffer = SmartBuffer.fromBuffer(tx.payload);
+                    let length = VarInt.toNumber(buffer);
+                    let header = Utils.readBuffer(buffer, length);
+
+                    switch (header.toString()) {
+                        case ProposalFeeData.HEADER: {
+                            let buffer = SmartBuffer.fromBuffer(tx.payload);
+                            let proposal_fee = ProposalFeeData.deserialize(buffer);
+                            await save_proposal_fee(storage, block_height, tx_hash, proposal_fee.proposal_id)
+                            await save_payload(storage, tx_hash, tx);
+                            break;
+                        }
+                        case ProposalData.HEADER:
+                            {
+                                let buffer = SmartBuffer.fromBuffer(tx.payload);
+                                let proposalData: ProposalData = ProposalData.deserialize(buffer);
+                                await save_proposal_data(storage, block_height, tx_hash, proposalData);
+                                await save_payload(storage, tx_hash, tx);
+                                break;
+                            }
+                        case BallotData.HEADER:
+                            {
+                                //TODO
+                                break;
+                            }
+                        default: {
+                            await save_payload(storage, tx_hash, tx);
+                            break;
+                        }
+                    }
+                    resolve();
+                }
+                catch (err) {
+                    reject(err);
```

Comment:
This is part of the process that should not be missed.
You have to report errors in detail using a logger.